docs/DeveloperGuide.md

```diff
@@ -2,37 +2,254 @@
+
+`storePatientInfo()` is used to store the patient information into the hashmap. 
+`getPatientInfo()` is used to retrieve the patient information from the hashmap.
+`printDiagnosis()` is used to print the diagnosis of the patient.
+`resetDiagnosticHistory()` is used to reset and clear the diagnostic history of the patient.
+`checkHash()` is used to check if the hashed password is correct and matches.
+`getAllPatientData()` is used to retrieve all the patient data from the hashmap.
+`resetSymptomChoice()` is used to reset the symptom choice of the patient in case an error was made.
+`hashPassword()` is used to hash the password of the patient.
+
+### Parser
+The parser class is responsible for parsing the user input. It is responsible for parsing the user input from the
+menu and the symptom list. It is also responsible for parsing the user input from the medicine list.
+
+`parseWelcome()` is used to parse the user input from the welcome menu.
+`parseAccountCommand()` is used to parse the user command and then execute the command based on the choice number.
+
+[comment]: <> (//@@author Brennanzuz)
+### Diagnosis
+
+![DiagnosisClassDiagram.png](diagrams%2FDiagnosisClassDiagram.png)
+
+#### Symptoms
+
+All possible symptoms that patients may suffer from are included into this enumeration.
+
+#### Illness
+
+This class is inherited by all other illnesses. It contains the name of the illness `illnessName`, its `severityLevel`,
+whether it is chronic `isChronic`, and most importantly, the list of symptoms specific to the illness `symptoms`.
+
+Child classes of `Illness` are given their values on creation of the object in their constructor function.
+Hence `new Diarrhoea()` will create an object with `.getSymptoms()` returning an `ArrayList<Symptom>`
+of `Symptom.WET_STOOL`, `Symptom.STOMACH_ACHE`, and `Symptom.DIARRHOEA`.
+
+Knowing this, each child object of `Illness` will be created and stored in `ArrayList<Illness> ALL_ILLNESSES`.
+
+#### Diagnosis algorithm
+
+![DiagnosisSequenceDiagram.png](diagrams%2FDiagnosisSequenceDiagram.png)
+
+After the patient has entered all their symptoms, it will be passed to the `getPossibleIllnesses()` function
+in `Diagnosis` where it will match each symptom the patient has to each symptom of all possible
+illnesses `ALL_ILLNESSES`.
+
+The percentage of matches the patient's symptoms has to the total number of symptoms the illness is specified to exhibit
+is calculated, and if this percentage does not cross the `POSSIBILITY_THRESHOLD`, the patient would be declared to
+possibly have the illness. To store this likely candidate, an `IllnessMatch` object is created storing the
+candidate `Illness` and the `similarityPercentage`.
+
+An `ArrayList<IllnessMatch> possibleIllnesses` is then returned to retrieve the appropriate medicine in another class.
+
+### Implementation of MedicineManager class, Medicine class
+
+![img_1.png](diagrams%2FMedicineSequenceDiagram.png)
```

Comment:
Perhaps the new Medicine should include the constructor of the Medicine class and classes in the sequence diagram should include colons.